graph/path/bellman_ford_moore.go

```diff
@@ -41,8 +41,8 @@ func BellmanFordFrom(u graph.Node, g graph.Graph) (path Shortest, ok bool) {
 	// the worst case complexity.
 	// If the queue-loop has more iterations than the amount of maximum edges
 	// it indicates that we have a negative cycle.
-	maxEdges := len(nodes) * (len(nodes) - 1)
-	var loops int
+	maxEdges := int64(len(nodes)) * int64(len(nodes)-1)
```

Comment:
The alternative for the negative cycle detection in Bellman-Ford is to use division at the condition below and keep it in `int`. I chose this approach because it has zero performance impact on amd64 while a division per search loop would be non-zero.